### ISSUE TITLE
slickGrid column resizing not worked Properly in RTL  

### ISSUE THREAD

**Issue by MaziarSangi** [6pac/SlickGrid#446]
Hello I Used SlickGrid in RTL Theme and When I resize the column width it wont resize the column in right way.

**Comment by 6pac** [comment]
You have just changed repos, so firstly just checking you are now using our repo's codebase.

**Comment by ghiscoding** [comment]
You might need [this](https://stackoverflow.com/questions/218065/overflow-to-left-instead-of-right/39874526#39874526) `direction: rtl` as pointed out in this [SO(https://stackoverflow.com/a/19738924/1212166) question.

**Comment by NadavPopplewell** [comment]
I have set direction : rtl on the body tab,
the slickgrid columns display exactly the same as in LTR mode,
but the header  row is misaligned (too much to the left)
seems to be a problem with 
<div class="slick-header-columns ui-sortable" style="left: -1000px;width: 2141px;direction: rtl;" unselectable="on">

What is this 'RTL theme' that the first poster refers to?

thanks,
Nadav

**Comment by 6pac** [comment]
I suspect they just use ```direction: rtl``` as @ghiscoding said. We've never built in or tested RTL support explicitly in SlickGrid, but it's all just HTML. If you want to tweak and test, we're happy to incorporate any changes or alternate CSS.

**Comment by NadavPopplewell** [comment]
But it's not only CSS, is it?
There seems to be some logic in the javascript code that adds left:+/-1000px to the header 
This seems to cause problems when the style is direction:RTL

**Comment by 6pac** [comment]
https://stackoverflow.com/questions/25172683/why-does-slickgrid-add-1000px-in-js-then-reverse-it-in-css-style

I'll help if I can but I have never used RTL languages, so I don't think I'm the best person to be testing that scenario. If you could provide a new RTL example, I'm happy to tweak elements of the grid where necessary to support the example in working correctly. Possibly RTL will be a new Slickgrid ```option```?

**Comment by NadavPopplewell** [comment]
How do I add the RTL example?

**Comment by 6pac** [comment]
Start with an existing example page (you might be best with a fairly fully featured one, like 'model', and you might want to check additional features like grouping and have more than one example page).  Just call them 'example4-model-RTL', for example.  
You can create a PR to add them, or just tell me which ones and post the basic mods to the source and I'll add them to the repo. Then you can play with them and create a PR if you like, or just post code snippets until we're happy with the result, then I'll commit.

**Comment by MorShaul** [comment]
can you help me to find where in the code is there a reference to resize columns (by dragging header columns), it doesn't work right in RTL

**Comment by 6pac** [comment]
All the resize column code is in ```setupColumnResize()``` in ```slick.grid.js```. It's possible that it's only set up for right handed column adjustment, but I don't know how RTL affects CSS. For example, I'm not sure if ```.append()``` appends to the left under RTL. That would make sense.  So the drag handle setup is probably OK, but perhaps the adjustment code needs to be adjusted.

Rally, an example is needed so we can actually see how this works and play around with it.

**Comment by ghiscoding** [comment]
So I took another look at this and all you need is to create a CSS class named "rtl" and then assign it to both `headerCssClass` (for the header titles) and `cssClass` (for the cells) and that should be enough

```html
<style>
.rtl {
  direction: rtl;
}
</style>
<script>
var columns = [
  {id: "title", name: "Title", field: "title", cssClass: "rtl", headerCssClass: "rtl"},
];
</script>
```
in our next v5, we'll add CSS variables  to our default them so that you can change the CSS to `display: inline-flex` and also `justify-content` to `flex-start` (or `flex-end`) BUT going this route will require the user to wrap the text in a span because doesn't work with `display: flex`. 

### Second Option (already doable)
Technically speaking, you could already do that yourself with the current styling theme, just modify the CSS

```css
<style>
.slick-header-column, 
.slick-cell {
  display: inline-flex;
  justify-content: flex-end;
}
.slick-header-column .slick-column-name,
.slick-cell span.child-cell {
  display: block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}
</style>
<script>
// and then use the same css classes
var columns = [
  {id: "title", name: "Title", field: "title", cssClass: "child-cell"},
];
</script>
```

**Comment by ghiscoding** [comment]
RTL is now properly supported, thanks to @jahanbakhsh18 for improving it, the fixes are now available in [v5.19.0](https://github.com/6pac/SlickGrid/releases/tag/5.19.0), there's also a new example linked in the release.
